### PR TITLE
Create a separate DecoupledRefereesController for the new decoupled references implementation

### PR DESCRIPTION
**Pull request by davidgisbey**
## Context

It makes sense to implement this flow completely independently so they codebase doesn't get too entwined.

See https://ukgovernmentdfe.slack.com/archives/CN1MCQCHZ/p1601459837044900

## Changes proposed in this pull request

- Create a new controller for the flow
- Move the relevant routes/views over

## Guidance to review

For now, I've just added decoupled_ before the controller/routes. These can all be removed once the flag is removed.

Any thoughts on naming would be great 👍 

## Link to Trello card

https://trello.com/c/npGvdYqK/2213-%F0%9F%92%94-dev-request-a-referee-flow

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/controllers/candidate_interface/decoupled_referees_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module CandidateInterface
+  class DecoupledRefereesController < CandidateInterfaceController
```
Comment (malcolmbaig) [inline app/controllers/candidate_interface/decoupled_referees_controller.rb:2]: The resource we're creating is a Reference, so maybe `DecoupledReferences`? Consistent with the way we're describing the feature too.

File: config/routes.rb [hunk 2]
```diff
@@ -393,6 +393,13 @@
         patch '/cancel/:id' => 'referees#cancel', as: :cancel_referee
       end
 
+      scope '/decoupled-referees' do
```
Comment (malcolmbaig) [inline config/routes.rb:396]: I like the prefixing of controllers and path helpers for clarity while this features is in-progress, but maybe the scope could just be `/references`? That way no visible change to any clients of this code when we eventually come back and remove the feature flag.

File: app/controllers/candidate_interface/decoupled_referees_controller.rb (hunk 1 above)
Comment (davidgisbey) [reply app/controllers/candidate_interface/decoupled_referees_controller.rb:2]: Yea i agree with this 👍

File: config/routes.rb (hunk 2 above)
Comment (davidgisbey) [reply config/routes.rb:396]: And also this. I'll make the changes now

**Review by malcolmbaig — APPROVED** [review]

**Review by tvararu — APPROVED** [review]
Happy to merge like this now and remove the `Decoupled` later :+1:

**Pull request merged**